### PR TITLE
Pm2 server deploy

### PR DESCRIPTION
**Pull request by philipep-galdino**
# Improve server deploy with pm2

_Make the server start along with the Indexer service restarting and watching out for the blockchain. Basically now they both start along with pm2._

## How should this be tested?

1. run pm2 restart ecosystem.config.js 
2. run pm2 logs
3. check if the applications are running correctly

## Notes or observations

_I'm having a problem with pm2 on the dev server we have. I can not set it up there to run `pm2 restart ecosystem.config.js`, even after `yarn add global pm2` went well with no errors it says that pm2 command was not found. Is it going to run correctly when deploy? I wanted to test it first._

## Linked issues

closes #00, closes #001, closes https://github.com/EveripediaNetwork/issues/issues/123

### CODE REVIEW

**Comment by philipep-galdino** [comment]
Created directly from the last branch that we set up pm2 just because it's a do-along task, lol. 

check out my doubt over the PR description, please!

**Comment by kesar** [comment]
https://stackoverflow.com/questions/40317578/yarn-global-command-not-working make sure yarn is exported in zsh / bash

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -18,4 +18,4 @@ jobs:
           script: |
             cd ep-api
             git pull
-            sudo supervisorctl restart all
+            pm2 restart ecosystem.config.js
```
Comment (kesar) [inline .github/workflows/deploy.yml:21]: missing the yarn to get latest packages

**Review by kesar — CHANGES_REQUESTED** [review]
missing yarn install

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (philipep-galdino) [reply .github/workflows/deploy.yml:22]: yeep, forgot about it! haha done

**Review by kesar — APPROVED** [review]

**Comment by philipep-galdino** [comment]
so, basically what I'm trying to do is improve how the service runs in the background - pm2 doesn't have such thing as to 'keep it alive', as it by default does keeps the application alive. I ran it on my local and the application was running ok and the indexer service were running and being restarted every 15min. 

what I notice is that I can try to improve how the indexer service is going to be working as instead of setting it on and off every 15min, just keep it on as the application does and restart if anything wrong happens.

I did a little bash script there on the deploy to wait for the application - the error that you told me about indexer service trying to start, failing and then restarting was because of this, the application and the indexer were tried to run side by side when the indexer needs the application on first. 

I still have to try it out on the server itself, tho.

File: indexer-ecosystem.config.js [hunk 2]
```diff
@@ -1,10 +1,5 @@
 module.exports = {
   apps: [
-    {
```
Comment (kesar) [inline indexer-ecosystem.config.js]: why remove it from here?

**Comment by kesar** [comment]
> I did a little bash script there on the deploy to wait for the application - the error that you told me about indexer service trying to start, failing and then restarting was because of this, the application and the indexer were tried to run side by side when the indexer needs the application on first.

Why indexer needs api running? I've been running indexer without API if I dont recall wrong

You need to add --loop true to the indexer to keep it looping without restarting

**Comment by philipep-galdino** [comment]
> Why indexer needs api running? I've been running indexer without API if I dont recall wrong
> 
> You need to add --loop true to the indexer to keep it looping without restarting

ok, got it. I knew I was forgetting something and there was a overengineering in this lol. 😅  and yeah I totally thought that the indexer was somehow depended to the API. 

I guess I can do it simpler then, let me push the changes

**Comment by philipep-galdino** [comment]
managed to set yarn global to work on the server. yarn global bin wasn't set, that was the problem. working fine now!

**Comment by philipep-galdino** [comment]
all done! 

to test all of this;
`pm2 start ecosystem.config.js --only ep-api`
`pm2 status ep-api` then if ready
`pm2 start ecosystem.config.js --only indexer-service`

to check out the whole service running:
`pm2 logs` 
and you'll get both ep-api running along with the indexer-service catching IFPS hashes and logging it out (if there are anym, if not, it'll just keep running until it gets)

**Review by kesar — APPROVED** [review]

**Pull request merged**